### PR TITLE
sys/hashes: Add Fletcher's checksums

### PR DESCRIPTION
**Pull request by gebart**
Adds Fletcher's checksum implementations by an anonymous contributor on Wikipedia.

I have been using these functions in hobby projects for many years on ARM, Atmega8 and AVR32 platforms, but I have not performed any formal tests of them other than that they seem to work fine for detecting lost bytes or other transmission errors when using UART to dump data to the PC.

### CODE REVIEW

File: sys/include/hashes.h [hunk 1]
```diff
@@ -17,10 +17,11 @@
  *
  * @author      Jason Linehan <patientulysses@gmail.com>
  * @author      Christian Mehlis <mehlis@inf.fu-berlin.de>
+ * @author      Joakim Gebart <joakim.gebart@eistec.se>
  */
 
-#ifndef __HASHES_H
-#define __HASHES_H
+#ifndef HASHES_H_
+#define HASHES_H_
```
Comment (LudwigOrtmann) [inline sys/include/hashes.h]: please make this a separate commit

**Comment by LudwigOrtmann** [comment]
please add a unittest

**Comment by gebart** [comment]
Rebased, split the commit into two, didn't have the energy to write a unit test.

**Comment by OlegHahm** [comment]
Of course unittests would be nice, but since we don't have neither unittests for already included hash functions nor coding conventions (yet) that require new code to come along with unittests, I think it's fine without.

**Comment by authmillenon** [comment]
@OlegHahm was that an ACK?

**Comment by OlegHahm** [comment]
No.

**Comment by OlegHahm** [comment]
Haven't neither reviewed nor tested.

**Comment by LudwigOrtmann** [comment]
Do you have any reference hashes?

**Comment by LudwigOrtmann** [comment]
@gebart Can you point to some reference hash sums?

**Comment by gebart** [comment]
@LudwigOrtmann I found some old measurement data from another project in a file, I don't have possibility to test it right now, but the file confirmed as having correct checksums when it was saved.

In hexadecimal (23 chars):
```
18b6 8605 0000 0000 0100 0000 04e3 1155 eb29 7105 001a e4
```
should have the checksum `0x3480` (little endian if I read the code in my old scripts correctly).

Another packet from the same measurement (also 23 chars):
```
86db 8608 0000 0000 4c89 0100 37e3 1155 f644 c102 210d e4
```
should have the checksum `0x5bc7`.

**Comment by gebart** [comment]
Below is a naive Python implementation of fletcher16 that I have used before:

```python
def fletcher16(data):
    """
    Compute Fletcher's checksum (16 bit variant) of the given string.
    
    @param[in] data Input string
    @return the computed checksum as an integer
    """
    sum1 = 0
    sum2 = 0
    length = len(data)
    for pos in range(length):
        sum1 = (sum1 + data[pos]) % 255
        sum2 = (sum2 + sum1) % 255

    if sum1 == 0x00:
        sum1 = 0xFF
    if sum2 == 0x00:
        sum2 = 0xFF
    return (sum2 << 8) | sum1
```

**Comment by josephnoir** [comment]
I checked the fletcher16 implementation with your data and it works! Code looks good too.

**Comment by gebart** [comment]
Is that an ACK?

**Comment by josephnoir** [comment]
I would prefer tests, but otherwise ... yes: ACK.

**Comment by gebart** [comment]
@josephnoir can you add your small test as a follow-up PR?
